src/storage.cpp

```diff
@@ -1008,18 +995,33 @@ namespace libtorrent
 		print_open_files("release files", m_files.name().c_str());
 #endif
 
-		for (std::map<std::string, int>::const_iterator i = to_move.begin()
-			, end(to_move.end()); i != end; ++i)
+        for (int i = 0; i < f.num_files(); ++i)
 		{
-			std::string old_path = combine_path(m_save_path, i->first);
-			std::string new_path = combine_path(save_path, i->first);
+            // files moved out to absolute paths are not moved
+            if (is_complete(f.file_path(i))) continue;
+
+            std::string old_path = combine_path(m_save_path, f.file_path(i));
+            std::string new_path = combine_path(save_path, f.file_path(i));
 
 			e.clear();
-			rename(old_path, new_path, e);
-			// if the source file doesn't exist. That's not a problem
-			// we just ignore that file
-			if (e == boost::system::errc::no_such_file_or_directory)
-				e.clear();
+            if (!exists(old_path))
+                continue;
+
+            if (has_parent_path(new_path))
+                create_directories(parent_path(new_path), e);
```

Comment:
the error handling here seems a bit sloppy. should failure to create the directory really just be ignored? Any file moved prior to this one presumably needs to be moved back here if there's any failure, no?